From 84b6c38dc40eaced937778104f70ec8732498a6f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 21 Sep 2020 17:57:45 +0300
Subject: [PATCH] [SW-20141] gaudi: (re)enable the pcie access

During the new reset sequence, we disable the PCIe access before the
hard reset and we should reenable it after the reset has been performed.
Meaning we do this only if the PCIe link is already up.
Enable the PCIe access after reset.

Change-Id: Ie11e619b768be998f311974c2f1ef30bb818edab
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 1e509d4..8a6c03c 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -14,6 +14,7 @@
 #include <cpu_timestamp_regs.h>
 
 #include <pcie_aux_masks.h>
+#include <pcie_dbi_masks.h>
 
 #include <hl_boot_if.h>
 #include <gaudi/gaudi_reg_map.h>
@@ -161,6 +162,19 @@
 	return reset_cause;
 }
 
+void ppboot_pcie_enable_access(void)
+{
+	u32 reg;
+
+	/* Enable the ELBI access */
+	regmap_writel(0, mmPCIE_CORE_REQ_RETRY_EN);
+
+	/* Enable the PCIe inbound memory requests */
+	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
+	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
+	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
+}
+
 /*
  * This logic copied from BTL PCIe section (#4)
  * In BTL it is called when DIP switch for PCIe enable is set
diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index 928ece2..d05fa03 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -112,11 +112,13 @@
 
 __weak u32 preboot_clr_flops_reset_cause(void) { return 0; }
 __weak void ppboot_log_waiting_for_pcie(void) {}
+__weak void ppboot_pcie_enable_access(void) {}
 
 static void ppboot_pcie_init(void)
 {
 	if (hl_pcie_link_up()) {
 		preboot_clr_flops_reset_cause();
+		ppboot_pcie_enable_access();
 		preboot_puts("PCIe link is already UP!\n");
 		return;
 	}
